src/main/java/org/elasticsearch/watcher/ResourceWatcherService.java

```diff
@@ -78,15 +78,21 @@ private Frequency(TimeValue interval) {
     @Inject
     public ResourceWatcherService(Settings settings, ThreadPool threadPool) {
         super(settings);
-        this.enabled = settings.getAsBoolean("watcher.enabled", true);
+        this.enabled = settings.getAsBoolean("resource.reload.enabled", true);
         this.threadPool = threadPool;
 
-        TimeValue interval = settings.getAsTime("watcher.interval.low", Frequency.LOW.interval);
+        TimeValue interval = settings.getAsTime("resource.reload.interval.low", Frequency.LOW.interval);
         lowMonitor = new ResourceMonitor(interval, Frequency.LOW);
-        interval = settings.getAsTime("watcher.interval.medium", settings.getAsTime("watcher.interval", Frequency.MEDIUM.interval));
+        interval = settings.getAsTime("resource.reload.interval.medium", settings.getAsTime("resource.reload.interval", Frequency.MEDIUM.interval));
         mediumMonitor = new ResourceMonitor(interval, Frequency.MEDIUM);
-        interval = settings.getAsTime("watcher.interval.high", Frequency.HIGH.interval);
+        interval = settings.getAsTime("resource.reload.interval.high", Frequency.HIGH.interval);
         highMonitor = new ResourceMonitor(interval, Frequency.HIGH);
+
+        logDeprecatedSetting("watcher.enabled", "resource.reload.enabled");
```

Comment:
Is this really a deprecation, or should it be an error? Deprecation to me implies the old settings still work, but it seems like above only the new setting works?